### ISSUE TITLE
Re-using the same model in a EpoxyController

### ISSUE THREAD

**Issue by luxw** [airbnb/epoxy#655]
My first time using Epoxy, and I'm having a little bit of trouble with re-using models.

I have a list of things, separated by a divider and then the list goes on
The difference between above/below the divider is a property of the object behind the model, some styling and a different click listener.

E.g. 
Item1
--------- (divider)
~~Item2~~

Clicking on Item1 makes it go down the divider and appear crossed and vice-versa.
However if I re-use the same EpoxyModel, the styling and clickListener does not change according to my model build code.

Then my list will behave strangely and this happens if I click on Item2
Item1
~~Item2~~
--------- (divider)

and Item2 clickListener would not change to `viewModel.moveDown()`

```kotlin
   override fun buildModels(items: List<MyItem>) {
        items
            .filter { !it.prop }
            .forEach {
                listItem {
                    id(it.uid)
                    item(it)
                    onClickListener { model, _, _, _ ->
                        viewModel.moveDown(model.item())
                    }
                }
            }

       listDivider {
            id("divider")
        }

        items
            .filter { it.prop }
            .forEach {
                listItem {
                    id(it.uid)
                    item(it)
                    onBind { _, view, _ ->
                        view.dataBinding.root.item_name.strikeText()
                    }
                    onClickListener { model, _, _, _ ->
                        viewModel.moveUp(model.item()) }
                }
            }
    }
```

If I create a different layout file for the "crossed" item, and use it as a different model, then it works just fine.

The data source is a LiveData backed by Room. 
`moveUp` and `moveDown` updates this field in the DB and triggers the live data.
When live data is triggered I use `setData()`.
Also any suggestions or things I'm doing wrong are hugely appreciated.
Thank you.

Edit: Disabling enableDoNotHash fixes the click listener, but not the styling.

**Comment by elihart** [comment]
```
onBind { _, view, _ ->
                        view.dataBinding.root.item_name.strikeText()
                    }
```
is not the right way to modify a view - all view modifications should be done through properties on the model. eg
```
listItem {
                    id(it.uid)
                    item(it)
                   strikeThrough(true)
```


for more information on the click handling, see the wiki https://github.com/airbnb/epoxy/wiki/DoNotHash

in general you click listener shouldn't change behavior - in this case it can simply say that an item was clicked, and separate logic an look at the current state of your data to determine whether it should be moved up or down

**Comment by luxw** [comment]
I did these changes you mentioned

First, I created a BindingAdapter for striking through the text and added a boolean variable to my layout. 
```
list item {
    id(it.uid)
    item(it)
    strikeThrough(true)
}
```

The top listItem has strikeThrough(false) and the bottom, true.
Then, I made changes to the viewModel function so `viewModel.move()` moves up or down based on the data property.

But nothing changed. Same problem as before. 
Also I have no access to `onModelClickListener`, should I be using that?

Thanks
